cmd/nri-kubernetes/main.go

```diff
@@ -49,59 +48,92 @@ func main() {
 		os.Exit(ExitIntegration)
 	}
 
-	// TODO: Here we will switch-case between components: KSM, ControlPlane, etc.
-	err = runKSM(&conf, clients, i)
-	if err != nil {
-		logger.Errorf(err.Error())
-		os.Exit(ExitLoop)
+	var ksmScraper *ksm.Scraper
+	if c.KSM.Enabled {
+		ksmScraper, err = setupKSM(c, clients)
+		if err != nil {
+			logger.Errorf("setting up ksm scraper: %w", err)
+			os.Exit(ExitSetup)
+		}
+		defer ksmScraper.Close()
+	}
+
+	for {
+		// TODO think carefully to the signature of this function
+		err := runScrapers(c, ksmScraper, i, clients)
+		if err != nil {
+			logger.Errorf("retrieving scraper data: %v", err)
+			os.Exit(ExitLoop)
+		}
+
+		err = i.Publish()
+		if err != nil {
+			logger.Errorf("publishing integration: %v", err)
+			os.Exit(ExitLoop)
+		}
+
+		time.Sleep(c.Interval)
 	}
 }
 
-func buildClients() (*clusterClients, error) {
-	k8s, err := client.NewKubernetes(true)
-	if err != nil {
-		return nil, fmt.Errorf("building kubernetes client: %w", err)
+func runScrapers(c config.Mock, ksmScraper *ksm.Scraper, i *integration.Integration, clients *clusterClients) error {
```

Comment:
I wonder how this structure will end up being after implementing commands and subcommands as @alvarocabanas and @carlossscastro  is doing. But for the moment they look fine!

Reply:
yes this is still work in progress that will addressed in the next PR when merging Alvaro and Carlos work